zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -200,6 +211,10 @@ protected ZkClient(IZkConnection zkConnection, int connectionTimeout, long opera
     if (zkConnection == null) {
       throw new NullPointerException("Zookeeper connection is null!");
     }
+
+    String syncUpOnNewSession = System.getProperty(ZkSystemPropertyKeys.ZK_AUTOSYNC_ENABLED, ZK_AUTOSYNC_ENABLED_DEFAULT);
+    _syncOnNewSession = Boolean.parseBoolean(syncUpOnNewSession);
```

Comment:
Should this config per zkClient or globally in one JVM? If it is globally, the syncUpOnNewSession should be static and initiated at place it is defined?

Reply:
Otherwise, if it is per-zkClient, should we add this as one parameter in the constructor?

Reply:
For now, it is globally unique. _syncOnNewSession as instance variable can help the change later if we want to make it instance configurable. 

Anyway, let me change it to global unique since huizhi also point it out.

Reply:
fixed.